src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -86,14 +112,7 @@ module.exports = function<T, P, I, C>(config : HostConfig<T, P, I, C>) {
       var ctor = workInProgress.type;
       workInProgress.stateNode = instance = new ctor(props);
     } else if (typeof instance.shouldComponentUpdate === 'function') {
-      if (current && current.memoizedProps) {
-        // Revert to the last flushed props, incase we aborted an update.
-        instance.props = current.memoizedProps;
-        if (!instance.shouldComponentUpdate(props)) {
-          return bailoutOnCurrent(current, workInProgress);
-        }
-      }
-      if (!workInProgress.childInProgress && workInProgress.memoizedProps) {
+      if (workInProgress.memoizedProps !== null) {
```

Comment:
This conditional should swap too then, I guess. I suspect I'll have to break this out when we add more composite life-cycle stuff here.